dolphin/api/v1/storages.py

```diff
@@ -42,11 +48,55 @@ def validate_parameters(data, required_parameters,
             msg = _("Required parameter %s is empty.") % parameter
             raise exc_response(explanation=msg)
 
+LOG = log.getLogger(__name__)
+
+
+def build_storages(storages):
+    # Build list of storages
+    views = [build_storage(storage)
+             for storage in storages]
+    return dict(storages=views)
+
+
+def build_storage(storage):
+    view = copy.deepcopy(storage)
+    return view
+
 
 class StorageController(wsgi.Controller):
 
     def index(self, req):
-        return dict(name="Storage 1")
+
+        supported_filters = [
+            'name',
+            'vendor',
+            'model',
+            'status',
+        ]
+        query_params = {}
+        query_params.update(req.GET)
+        # update options  other than filters
+        sort_keys = (lambda x: [x] if x is not None else x)(query_params.get('sort_key'))
+        sort_dirs = (lambda x: [x] if x is not None else x)(query_params.get('sort_dir'))
+        limit = query_params.get('limit', None)
+        offset = query_params.get('offset', None)
+        # strip out options except supported filter options
```

Comment:
need concern about the marker

Reply:
added marker reading here, though now we dont use this db adapter now.